tests/scalers_go/custom_hpa_name/custom_hpa_name_test.go

```diff
@@ -0,0 +1,178 @@
+  minReplicaCount: 1
+  maxReplicaCount: 1
+  cooldownPeriod: 10
+  advanced:
+    horizontalPodAutoscalerConfig:
+      name: {{.CustomHpaName}}
+  triggers:
+      - type: cpu
+        metadata:
+          type: Utilization
+          value: "50"
+`
+)
+
+func TestCustomToDefault(t *testing.T) {
+	// setup
+	testName := "custom-to-default-hpa-name"
+	scaledObjectName := fmt.Sprintf("%s-so", testName)
+	defaultHpaName := fmt.Sprintf("keda-hpa-%s", scaledObjectName)
+	customHpaName := fmt.Sprintf("%s-custom", testName)
+	test(t, testName, customHpaName, scaledObjectTemplateWithCustomName, "custom",
+		defaultHpaName, scaledObjectTemplate, "default")
+}
+
+func TestDefaultToCustom(t *testing.T) {
+	// setup
+	testName := "default-to-custom-hpa-name"
+	scaledObjectName := fmt.Sprintf("%s-so", testName)
+	defaultHpaName := fmt.Sprintf("keda-hpa-%s", scaledObjectName)
+	customHpaName := fmt.Sprintf("%s-custom", testName)
+	test(t, testName, defaultHpaName, scaledObjectTemplate, "default",
+		customHpaName, scaledObjectTemplateWithCustomName, "custom")
+}
+
+func test(t *testing.T, testName string, firstHpaName string, firstSOTemplate string, firstHpaDescription string,
+	secondHpaName string, secondSOTemplate string, secondHpaDescription string) {
+	testNamespace := fmt.Sprintf("%s-ns", testName)
+	deploymentName := fmt.Sprintf("%s-deployment", testName)
+	scaledObjectName := fmt.Sprintf("%s-so", testName)
+	customHpaName := fmt.Sprintf("%s-custom", testName)
+	// Create kubernetes resources
+	kc := GetKubernetesClient(t)
+	data := getTemplateData(testNamespace, deploymentName, scaledObjectName, customHpaName)
+	templates := []string{deploymentTemplate, firstSOTemplate}
+
+	CreateKubernetesResources(t, kc, testNamespace, data, templates...)
+
+	t.Logf("--- validate hpa is with %s name ---", firstHpaDescription)
+	hpa, _ := WaitForHpaCreation(t, kc, firstHpaName, testNamespace, 10, 1)
+	assert.Equal(t, firstHpaName, hpa.Name)
+
+	t.Log("--- change hpa name ---")
+	templatesCustomName := []string{secondSOTemplate}
+	KubectlApplyMultipleWithTemplate(t, data, templatesCustomName...)
+
+	t.Logf("--- validate new hpa is with %s name ---", secondHpaDescription)
+	hpa, _ = WaitForHpaCreation(t, kc, secondHpaName, testNamespace, 10, 1)
+	assert.Equal(t, secondHpaName, hpa.Name)
+
+	t.Logf("--- validate hpa with %s name does not exists ---", firstHpaDescription)
+	_, err := WaitForHpaCreation(t, kc, firstHpaName, testNamespace, 10, 1)
+	assert.True(t, errors.IsNotFound(err))
+
+	// cleanup
+	DeleteKubernetesResources(t, kc, testNamespace, data, templates...)
+}
+
+func getTemplateData(testNamespace string, deploymentName string, scaledObjectName string, customHpaName string) templateData {
+	return templateData{
+		TestNamespace:    testNamespace,
+		DeploymentName:   deploymentName,
+		ScaledObjectName: scaledObjectName,
+		CustomHpaName:    customHpaName,
+	}
+}
+
+func WaitForHpaCreation(t *testing.T, kc *kubernetes.Clientset, name, namespace string,
```

Comment:
Should this be a helper? I mean, maybe other tests requires it in the future

Reply:
Wondered that myself. 
In the current e2e tests (even the tests that still written in typescript) it's not necessary, and that's why I didn't put it the for now

You think it should be in helper? I can move it to there